dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
I suggest using constexpr here. That way we get all of the benefits of an
enum (e.g. being able to use it as a template parameter), without
introducing a superfluous enum.

(Although, in this particular case, it's pretty unlikely that we will ever
do that.)